From f7aaa72602de2e77f29d650a91b8a844d1899d2b Mon Sep 17 00:00:00 2001
From: Seth Hillbrand <seth@kipro-pcb.com>
Date: Tue, 28 Jul 2026 07:39:06 -0700
Subject: [PATCH 2/2] Rename swig functions for 4.5 compat

Newer swig removed Python 2 compatibility macros that we were relying
on.  This patch is from the Fedora team.  It removes the macro usage and
replaces it with the underlying function

(cherry picked from commit 2149e5e2333e7edf71f92e41bf39a70e5849c490)
---
 common/swig/math.i |  4 ++--
 common/swig/wx.i   | 18 +++++++++---------
 2 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/common/swig/math.i b/common/swig/math.i
index 6f4f240897..8dc3cbd0ac 100644
--- a/common/swig/math.i
+++ b/common/swig/math.i
@@ -64,8 +64,8 @@
     PyObject* Get()
     {
         PyObject* tup = PyTuple_New(2);
-        PyTuple_SET_ITEM(tup, 0, PyInt_FromLong(self->x));
-        PyTuple_SET_ITEM(tup, 1, PyInt_FromLong(self->y));
+        PyTuple_SET_ITEM(tup, 0, PyLong_FromLong(self->x));
+        PyTuple_SET_ITEM(tup, 1, PyLong_FromLong(self->y));
         return tup;
     }
 
diff --git a/common/swig/wx.i b/common/swig/wx.i
index 464ce1770b..0f8fc4629e 100644
--- a/common/swig/wx.i
+++ b/common/swig/wx.i
@@ -79,10 +79,10 @@ public:
        PyObject* Get()
        {
             PyObject* res = PyTuple_New(4);
-            PyTuple_SET_ITEM(res, 0, PyInt_FromLong(self->x));
-            PyTuple_SET_ITEM(res, 1, PyInt_FromLong(self->y));
-            PyTuple_SET_ITEM(res, 2, PyInt_FromLong(self->width));
-            PyTuple_SET_ITEM(res, 3, PyInt_FromLong(self->height));
+            PyTuple_SET_ITEM(res, 0, PyLong_FromLong(self->x));
+            PyTuple_SET_ITEM(res, 1, PyLong_FromLong(self->y));
+            PyTuple_SET_ITEM(res, 2, PyLong_FromLong(self->width));
+            PyTuple_SET_ITEM(res, 3, PyLong_FromLong(self->height));
             return res;
         }
     }
@@ -122,8 +122,8 @@ public:
        PyObject* Get()
        {
             PyObject* res = PyTuple_New(2);
-            PyTuple_SET_ITEM(res, 0, PyInt_FromLong(self->x));
-            PyTuple_SET_ITEM(res, 1, PyInt_FromLong(self->y));
+            PyTuple_SET_ITEM(res, 0, PyLong_FromLong(self->x));
+            PyTuple_SET_ITEM(res, 1, PyLong_FromLong(self->y));
             return res;
         }
     }
@@ -174,8 +174,8 @@ public:
         PyObject* Get()
         {
             PyObject* tup = PyTuple_New(2);
-            PyTuple_SET_ITEM(tup, 0, PyInt_FromLong(self->x));
-            PyTuple_SET_ITEM(tup, 1, PyInt_FromLong(self->y));
+            PyTuple_SET_ITEM(tup, 0, PyLong_FromLong(self->x));
+            PyTuple_SET_ITEM(tup, 1, PyLong_FromLong(self->y));
             return tup;
         }
     }
@@ -301,7 +301,7 @@ public:
 
 
 %typemap(typecheck, precedence=SWIG_TYPECHECK_POINTER) wxString& {
-    $1 = PyString_Check($input) || PyUnicode_Check($input);
+    $1 = PyBytes_Check($input) || PyUnicode_Check($input);
 }
 
 %apply wxString& { wxString* }
-- 
2.52.0

